lib/galaxy/model/database_object_names.py

```diff
@@ -0,0 +1,44 @@
+"""
+Naming convention and helper functions for generating names of database
+constraints and indexes.
+"""
+from typing import (
+    List,
+    Union,
+)
+
+from galaxy.util import listify
+
+# Naming convention applied to database constraints and indexes.
+# All except "ix" are conventions used by PostgreSQL by default.
+# We keep the "ix" template consistent with historical Galaxy usage.
+NAMING_CONVENTION = {
+    "pk": "%(table_name)s_pkey",
+    "fk": "%(table_name)s_%(column_0_name)s_fkey",
+    "uq": "%(table_name)s_%(column_0_name)s_key",
+    "ck": "%(table_name)s_%(column_0_name)s_check",
+    "ix": "ix_%(table_name)s_%(column_0_name)s",
+}
+
+
+def foreign_key(table_name: str, column_names: Union[str, List]) -> str:
```

Comment:
Nothing important, but I think something like `build_foreing_key_name(...)` and in general `build_x_name` will be better here to avoid any confusion when calling these naming utilities.

Reply:
I agonized over this :-) I chose to go with these for conciseness and because we'll always import them from a module that's named `database_object_names`. So, you get `database_object_names.foreign_key()` (if importing the module). It cannot be `foreign_key_name` because in revision scripts we always assign the value to variables with the same name, so you can't have `index_name = index_name()`. However, I don't have a preference here, so if `build_index_name()` appears to be more straightforward, then let's go with that. I'll make the change.